sympy/matrices/determinant.py

```diff
@@ -380,39 +385,13 @@ def _charpoly(M, x='lambda', simplify=_simplify):
     >>> M.charpoly(x).gen == x
     False
 
-    Notes
-    =====
-
-    The Samuelson-Berkowitz algorithm is used to compute
-    the characteristic polynomial efficiently and without any
-    division operations.  Thus the characteristic polynomial over any
-    commutative ring without zero divisors can be computed.
-
-    If the determinant det(x*I - M) can be found out easily as
-    in the case of an upper or a lower triangular matrix, then
-    instead of Samuelson-Berkowitz algorithm, eigenvalues are computed
-    and the characteristic polynomial with their help.
```

Comment:
It's probably worth keeping these optimisations.

Reply:
Sure, will do.